### PR TITLE
Coverage improvements

### PR DESCRIPTION
**Pull request by rusty1968**
Improvements:
- Do not panic in coverage library crate.
- Disassemble binary to better identify non covered code.

### CODE REVIEW

File: coverage/src/main.rs [hunk 1]
```diff
@@ -3,14 +3,41 @@
...
+    instructions: &[Instruction],
+) {
+    for function in partially_covered_functions {
+        let start_address = function.address;
+        let end_address = start_address + function.size;
+
+        // Filter instructions for the current function
+        let uncovered_instructions = instructions
+            .iter()
+            .filter(|instruction| {
```
Comment (rusty1968) [inline coverage/src/main.rs:29]: Need bitmap filtering.

File: coverage/src/disasm.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
+// Licensed under the Apache-2.0 license
+use std::process::Command;
+
+use caliptra_builder::run_cmd_stdout;
+
+const OBJDUMP: &str = "riscv64-unknown-elf-objdump";
+
+pub struct Instruction(usize, String);
```
Comment (korran) [inline coverage/src/disasm.rs:8]:
```
pub struct Instruction {
  address: usize,
  instruction: &str,
}
``` ?

Comment (rusty1968) [reply coverage/src/disasm.rs:8]: Ok.

Comment (rusty1968) [reply coverage/src/disasm.rs]: Done.

File: builder/src/lib.rs [hunk 3]
```diff
@@ -51,18 +51,21 @@ fn run_cmd(cmd: &mut Command) -> io::Result<()> {
...
+    if let (Some(stdin), Some(input)) = (child.stdin.take(), input) {
+        let mut writer = BufWriter::new(stdin);
+        writer.write_all(input)?;
+        writer.flush()?;
     }
     let out = child.wait_with_output()?;
     if out.status.success() {
         Ok(String::from_utf8_lossy(&out.stdout).into())
     } else {
+        println!("status ={:?}", out.status.code());
```
Comment (korran) [inline builder/src/lib.rs:68]: remove this?

File: builder/src/lib.rs [hunk 4]
```diff
@@ -51,18 +51,21 @@ fn run_cmd(cmd: &mut Command) -> io::Result<()> {
...
-fn run_cmd_stdout(cmd: &mut Command, input: Option<&[u8]>) -> io::Result<String> {
+pub fn run_cmd_stdout(cmd: &mut Command, input: Option<&[u8]>) -> io::Result<String> {
     cmd.stdin(Stdio::piped());
     cmd.stdout(Stdio::piped());
 
     let mut child = cmd.spawn()?;
-    if let (Some(mut stdin), Some(input)) = (child.stdin.take(), input) {
-        std::io::Write::write_all(&mut stdin, input)?;
+    if let (Some(stdin), Some(input)) = (child.stdin.take(), input) {
+        let mut writer = BufWriter::new(stdin);
```
Comment (korran) [inline builder/src/lib.rs:60]: Does this change actually do anything?

Comment (rusty1968) [reply builder/src/lib.rs:60]: No. It was a left-over attempt to fix the broke pipe.

File: builder/src/lib.rs (hunk 3 above)
Comment (rusty1968) [reply builder/src/lib.rs]: Reverted to original and just made the symbol public.

**Review by nquarton — APPROVED** [review]

File: ci.sh [hunk 5]
```diff
@@ -97,10 +97,11 @@ fi
 if task_enabled "test"; then
   echo Run tests
   if cargo nextest help > /dev/null 2>/dev/null; then
-    CALIPTRA_PREBUILT_FW_DIR="${fw_dir}" cargo nextest run --config "${EXTRA_CARGO_CONFIG}" --locked
+    CPTRA_COVERAGE_PATH=/tmp CALIPTRA_PREBUILT_FW_DIR="${fw_dir}" cargo nextest run --config "${EXTRA_CARGO_CONFIG}" --locked
```
[the remark this answers is not in the document]
Comment (rusty1968) [reply ci.sh:100]: We are running coverage in ci.sh. I can use another folder, but we are actually extracting coverage info in ci.sh.

**Review by korran — APPROVED** [review]

**Pull request merged**